File: leet_0506_relative_ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        temp = sorted(score, reverse=True)
        res = []
        for n in score:

            idx = temp.index(n)
            if idx == 0:
                res.append("Gold Medal")
            elif idx == 1:
                res.append("Silver Medal")
            elif idx == 2:
                res.append("Bronze Medal")
            else:
                res.append(str(idx + 1))

        return res

File: leet_0506_relative_ranks.go
func findRelativeRanks(score []int) []string {
    sorted := make([]int, len(score))
    copy(sorted, score)
    // sorted := score
    sort.Sort(sort.Reverse(sort.IntSlice(sorted)))
    res := []string{}
    for _, n := range score {
        idx := -1
        for i, val := range sorted {
            if val == n {
                idx = i
                break
            }
        }

        switch idx {
            case 0:
                res = append(res, "Gold Medal")
            case 1:
                res = append(res, "Silver Medal")
            case 2:
                res = append(res, "Bronze Medal")
            default:
                res = append(res, strconv.Itoa(idx + 1))
        }
    }

    return res
}
